quipucords/scanner/openshift/api.py

```diff
@@ -106,10 +106,20 @@ def _dynamic_client(self):
 
     @classmethod
     def with_config_info(
-        cls, *, host, protocol, port, ssl_verify: bool = True, **kwargs
+        cls,
+        *,
+        host,
+        protocol,
+        port,
+        ssl_verify: bool = True,
+        proxy_url: str = None,
+        **kwargs,
     ):
         """Initialize OpenShiftApi without providing a KubeConfig object."""
         host_uri = f"{protocol}://{host}:{port}"
+        proxies = {}
+        if proxy_url:
+            proxies[protocol] = "http://" + proxy_url
```

Comment:
Like with Ansible, this will never have `https` protocol in the string? I see this again in OpenShift and won't repeat a third time. Potentially mismatching on `proxies[protocol]` feels wrong to me. Could this instead be:

```suggestion
            proxies[protocol] =  f"{protocol}://{proxy_url}"
```

Or are there cases where `protocol` is neither `'http` nor `https`?

Reply:
forcing the protocol to match host protocol will not always work. one can setup an https proxy using using HTTP (and I think vice versa works as well, but I'm not sure).

I'd suggest going expecting the proxy_url to explicitly include the proper protocol (of the proxy, not of the source - that would be the protocol used to communicate with the source).